doc/contributing/coding-conventions.chapter.md

```diff
@@ -230,185 +230,43 @@ There are a few naming guidelines:
-
-**If it’s a _server_:**
-
-- **If it’s a web server:**
-
-  - `servers/http` (e.g. `apache-httpd`)
-
-- **If it’s an implementation of the X Windowing System:**
-
-  - `servers/x11` (e.g. `xorg` — this includes the client libraries and programs)
-
-- **Else:**
-
-  - `servers/misc`
-
-**If it’s a _desktop environment_:**
-
-- `desktops` (e.g. `kde`, `gnome`, `enlightenment`)
-
-**If it’s a _window manager_:**
-
-- `applications/window-managers` (e.g. `awesome`, `stumpwm`)
-
-**If it’s an _application_:**
-
-A (typically large) program with a distinct user interface, primarily used interactively.
-
-- **If it’s a _version management system_:**
-
-  - `applications/version-management` (e.g. `subversion`)
-
-- **If it’s a _terminal emulator_:**
-
-  - `applications/terminal-emulators` (e.g. `alacritty` or `rxvt` or `termite`)
-
-- **If it’s a _file manager_:**
-
-  - `applications/file-managers` (e.g. `mc` or `ranger` or `pcmanfm`)
-
-- **If it’s for _video playback / editing_:**
-
-  - `applications/video` (e.g. `vlc`)
-
-- **If it’s for _graphics viewing / editing_:**
-
-  - `applications/graphics` (e.g. `gimp`)
-
-- **If it’s for _networking_:**
-
-  - **If it’s a _mailreader_:**
-
-    - `applications/networking/mailreaders` (e.g. `thunderbird`)
-
-  - **If it’s a _newsreader_:**
-
-    - `applications/networking/newsreaders` (e.g. `pan`)
-
-  - **If it’s a _web browser_:**
-
-    - `applications/networking/browsers` (e.g. `firefox`)
-
-  - **Else:**
-
-    - `applications/networking/misc`
-
-- **Else:**
-
-  - `applications/misc`
-
-**If it’s _data_ (i.e., does not have a straight-forward executable semantics):**
-
-- **If it’s a _font_:**
-
-  - `data/fonts`
-
-- **If it’s an _icon theme_:**
-
-  - `data/icons`
-
-- **If it’s related to _SGML/XML processing_:**
-
-  - **If it’s an _XML DTD_:**
-
-    - `data/sgml+xml/schemas/xml-dtd` (e.g. `docbook`)
-
-  - **If it’s an _XSLT stylesheet_:**
-
-    (Okay, these are executable...)
+For example, `pkgs.libfoo` may be declared in `pkgs/unit/li/libfoo/package.nix` with a Nix expression as follows:
+```nix
+{ lib, stdenv, libbar, enableBaz ? false }:
+stdenv.mkDerivation {
+  pname = "libfoo";
+  version = "...";
 
-    - `data/sgml+xml/stylesheets/xslt` (e.g. `docbook-xsl`)
+  buildInputs = [ libbar ];
 
-- **If it’s a _theme_ for a _desktop environment_, a _window manager_ or a _display manager_:**
+  makeFlags = lib.optional enableBaz "BAZ=1";
 
-  - `data/themes`
+  # ...
+}
+```
 
-**If it’s a _game_:**
+This automatically declares `pkgs.libfoo` as `pkgs.callPackage pkgs/unit/li/libfoo/package.nix { }`.
 
-- `games`
+If `libfoo` doesn't compile with the default `libbar` version, and `pkgs.libbar_2` is available, you can change the default `libbar` version by adding a definition like this to `pkgs/top-level/all-packages.nix`:
+```nix
+  libfoo = callPackage ../unit/li/libfoo/package.nix {
+    libbar = libbar_2;
+  };
+```
 
-**Else:**
+### Existing hierarchy {#sec-hierarchy}
```

Comment:
I think it's fair to remove the current categorization stuff. What I'm missing here though is some information about when to use the old hierarchy